sdk/python/kfp/dsl/for_loop.py

```diff
@@ -246,7 +246,8 @@ def from_pipeline_channel(
     ) -> 'LoopArtifactArgument':
         """Creates a LoopArtifactArgument object from a PipelineArtifactChannel
         object."""
-        assert channel.is_artifact_list is True, 'Cannot iterate over single Artifacts.'
+        if not channel.is_artifact_list:
```

Comment:
Thanks, Kevin. Can we add a test for this?

Reply:
Done.